assets/app/scripts/controllers/pod.js

```diff
@@ -79,10 +81,35 @@ angular.module('openshiftConsole')
       watches.push(DataService.watch("builds", $scope, function(builds) {
         $scope.builds = builds.by("metadata.name");
         Logger.log("builds (subscribe)", $scope.builds);
-      }));      
+      }));
+
+      (function initLogs() {
+        angular.extend($scope, {
+          logs: [],
+          logsLoading: true,
+          canShowDownload: false,
+          initLogs: initLogs
+        });
+        DataService.watchLogStream("pods/log", $routeParams.pod, $scope, function(line) {
+          $scope.$apply(function() {
+            $scope.logs.push({text: line});
+            $scope.canShowDownload = true;
+          });
+        }, function() {
+          $scope.$apply(function() {
+            $scope.logsLoading = false;
+          });
+        })
```

Comment:
This would have to have a fn to let it know when to actually create the socket, I think:

```javascript

// stream(kind, name, context)
//   .message(fn)
//   .close(fn)
//   .error(fn)
//   .go()  <-- needs something at the end to tell it when to do its thing
```